llvm/lib/Passes/PassBuilder.cpp

```diff
@@ -528,6 +529,106 @@ PassBuilder::PassBuilder(TargetMachine *TM, PipelineTuningOptions PTO,
 #include "llvm/Passes/MachinePassRegistry.def"
     });
   }
+  auto parseLevelParam = [](StringRef P) -> Expected<OptimizationLevel> {
+    auto OptLevel = llvm::StringSwitch<std::optional<OptimizationLevel>>(P)
+                        .Case("O0", OptimizationLevel::O0)
+                        .Case("O1", OptimizationLevel::O1)
+                        .Case("O2", OptimizationLevel::O2)
+                        .Case("O3", OptimizationLevel::O3)
+                        .Case("Os", OptimizationLevel::Os)
+                        .Case("Oz", OptimizationLevel::Oz)
+                        .Default(std::nullopt);
+    if (!OptLevel)
+      return llvm::createStringError(llvm::inconvertibleErrorCode(),
+                                     "invalid optimization level '%s'",
+                                     P.str().c_str());
```

Comment:
for future reference: you probably want to avoid converting StringRef to std::string in most circumstances (as it will allocate memory). createStringError should be able to take StringRef